src/sql/workbench/parts/query/common/queryInput.ts

```diff
@@ -301,9 +301,13 @@ export class QueryInput extends EditorInput implements IEncodingSupport, IConnec
 		this.state.connected = false;
 	}
 
-	public onConnectCanceled(): void {
+	public onConnectCanceled(skipConnectedStateReset?: boolean): void {
```

Comment:
>skipConnectedStateReset [](start = 26, length = 23)

It looks like we're not calling this anywhere else but the connectionDialogService so I think it's fine to just get rid of the parameter and just reset if state.connecting is truthy